examples/prefect/training.py

```diff
@@ -0,0 +1,51 @@
+import numpy as np
+from prefect import flow, task
+from sklearn import base
+from sklearn.datasets import make_regression
+from sklearn.linear_model import LinearRegression
+from sklearn.model_selection import train_test_split
+
+
+@task
+def make_regression_data(
+    random_state: int, n_samples: int = 100, n_features: int = 1, noise: float = 0.2
+) -> tuple[np.ndarray, np.ndarray]:
+    X, y = make_regression(
+        n_samples=n_samples, n_features=n_features, noise=noise, random_state=random_state
+    )
+    return X, y
+
+
+@task
+def make_train_test_split(
+    X: np.ndarray, y: np.ndarray, random_state: int, test_size: float = 0.2
+) -> tuple[np.ndarray, np.ndarray, np.ndarray, np.ndarray]:
+    X_train, X_test, y_train, y_test = train_test_split(
+        X, y, test_size=test_size, random_state=random_state
+    )
+    return X_train, y_train, X_test, y_test
+
+
+@flow
```

Comment:
Do we ever need to prepare regression data outside of the flow afterwards? If not, we could think about inlining this into `prepare_regressor_and_test_data` below.

Is there any noticeable advantage of having it on the dashboard etc. in your opinion?

Reply:
I was kinda on the fence about this one and had it inline at first. 
It makes a nice dependency graph in the Prefect UI but we are also not demo'ing prefect and the difference is not that big. Factoring it back